frontend/model/contracts/shared/voting/proposals.js

```diff
@@ -44,7 +44,7 @@ export function notifyAndArchiveProposal ({ state, proposalHash, proposal, contr
 export function buildInvitationUrl (groupId: string, groupName: string, inviteSecret: string, creatorID?: string): string {
   const rootGetters = sbp('state/vuex/getters')
   const creatorUsername = creatorID && rootGetters.usernameFromID(creatorID)
-  return `${location.origin}/app/join#?${(new URLSearchParams({
+  return `${location.origin}/app/join?${(new URLSearchParams({
```

Comment:
@corrideat Need your review on this

Reply:
Actually, the hash symbol `#` was not needed since we saved all the parameters inside query which lies after `?`.

Reply:
(also, if for some reason it works without a `#`, that's a bug: the query must not contain secrets)

Reply:
@Silver-IT This PR needs to be closed while keeping this URL using a `#`.

The reason we're using a hash here is because the hash prevents the server from learning about what groups the user is joining, and also prevents the invite secret from being shared with the server.

Reply:
> That's wrong, the # is needed.

Why needed?
##
@corrideat, what do you mean by the following code? Why could it be the reason that we should keep using `#`?
I think it's just to get a query (groupId, secret, ...) and nothing else.
```
this.ephemeral.hash = new URLSearchParams(this.$route.hash.slice(1))
```

I used `query` instead of `hash`, and also updated `Join.vue`.
```
this.ephemeral.query = Object.assign({}, this.$route.query)
```
##
> The reason we're using a hash here is because the hash prevents the server from learning about what groups the user is joining, and also prevents the invite secret from being shared with the server.

@taoeffect, what's wrong if we use `query` instead of `hash`? And the format `/join#?...` doesn't make sense. I can't understand the intention of the code lies after `?`. Is it `hash`? or `query`? Why `#` and `?` are there all together?

Reply:
> the query must not contain secrets

The invite link is created, there is a `secret` inside. Why is it fine to contain `secret` in `hash`, and why not in `query`?

Reply:
> Why is it bug? I have updated all the related codes, and works fine all time.
> The invite link is created, there is a secret inside. Why is it fine to contain secret in hash, and why not in query?

I answered both of these questions in my reply above.

> The reason we're using a hash here is because the hash prevents the server from learning about what groups the user is joining, and also >>>> **prevents the invite secret from being shared with the server.** <<<

Regarding:

> Why # and ? are there all together?

I'm not sure, perhaps it's to make it easier for using with `URLSearchParams`?

Reply:
> If the invite secret is shared with the server then it's not so secret anymore and the server could use it to join the group.

So do you mean the first one is good in terms of secret and the second one is bad?
```
http://localhost:8000/app/join#?groupId=z9brRu3VXVfvXAHFxjuzphoq8mSsPt4Y8fVREeM5g9mPoSVEhFUG&groupName=Dreamers&secret=%5B%22edwards25519sha512batch%22%2Cnull%2C%224nGfJcEZjoVk7MOnPAZlKUBJeQbQthkK2YPAb%2BTXJwxpEIZfXdqJpTuQssNu0MrZUDbGsIrn5bOJdIODk5zRKw%3D%3D%22%5D
```
```
http://localhost:8000/app/join?groupId=z9brRu3VXVfvXAHFxjuzphoq8mSsPt4Y8fVREeM5g9mPoSVEhFUG&groupName=Dreamers&secret=%5B%22edwards25519sha512batch%22%2Cnull%2C%224nGfJcEZjoVk7MOnPAZlKUBJeQbQthkK2YPAb%2BTXJwxpEIZfXdqJpTuQssNu0MrZUDbGsIrn5bOJdIODk5zRKw%3D%3D%22%5D
```

And it's just the format change of invite link. There is no more changes. I can't understand what you mean by secret problem. What is the different on the server side when the user uses these two formats?

Reply:
> So do you mean the first one is good in terms of secret and the second one is bad?

Yes.

> What is the different on the server side when the user uses these two formats?

The first one sends all of that information to the server and the second one doesn't.

Reply:
Now, I think I understood what you mean, @taoeffect. So we should use `hash` instead of `query` for the security reason.

Reply:
Yep! 👍